ee/debug/checkups/intune_windows.go

```diff
@@ -0,0 +1,95 @@
+//go:build windows
+// +build windows
+
+package checkups
+
+import (
+	"archive/zip"
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+type intuneCheckup struct {
+	summary string
+}
+
+func (i *intuneCheckup) Name() string {
+	return "Intune"
+}
+
+func (i *intuneCheckup) Run(ctx context.Context, extraWriter io.Writer) error {
+	// Other areas of interest: https://learn.microsoft.com/en-us/mem/intune/remote-actions/collect-diagnostics
+
+	zipWriter := zip.NewWriter(extraWriter)
+	defer zipWriter.Close()
+
+	if err := agentLogs(zipWriter); err != nil {
+		i.summary += fmt.Sprintf("Failed to collect Intune agent logs: %v. ", err)
+	}
+
+	if err := installLogs(zipWriter); err != nil {
+		i.summary += fmt.Sprintf("Failed to collect Intune install logs: %v. ", err)
+	}
+
+	i.summary = strings.TrimSpace(i.summary)
+
+	return nil
+}
+
+func agentLogs(zipWriter *zip.Writer) error {
+	agentLogsPathPattern := filepath.Join(os.Getenv("SYSTEMROOT"), "ProgramData", "Microsoft", "IntuneManagementExtension", "Logs", "*")
```

Comment:
Do we want `*` or `*.log` here?

Reply:
https://learn.microsoft.com/en-us/mem/intune/remote-actions/collect-diagnostics says `%ProgramData%\Microsoft\IntuneManagementExtension\Logs\*.*` and I haven't gotten set up enough to be able to determine if we can narrow it down by extension further

Reply:
I guess okay. I'm slightly worried we'll find a 500meg crash dump. But sure!